bp_common/syn/vivado/tcl/vivado_synth.tcl

```diff
@@ -0,0 +1,111 @@
+set proj_name "blackparrot_test"
+
+# Genesys 2 - xc7k325tffg900-2
+# Artix 7 - xc7a200tfbg676-2
+create_project -force $proj_name ./$proj_name -part $::env(PART)
+
+if {[string equal [get_filesets -quiet sources_1] ""]} {
+  create_fileset -srcset sources_1
+}
+
+set fileset_obj [get_filesets sources_1]
+set f [split [string trim [read [open "flist.vcs" r]]] "\n"]
+set flist [list ]
+foreach x $f {
+  # If the item has a file to be added, it starts with $BP_*_DIR
+  if {[string match "\$" [string index $x 0]]} {
+    set env_var [string trimleft $x "\$"]
+    regexp {([A-Za-z_]*)} $env_var a
```

Comment:
An easier way to do this is to set tcl variables corresponding to the environment variables.

```
set BP_COMMON_DIR $::env(BP_COMMON_DIR)
```

Then it will autoexpand